src/TensorOneD.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+
+#include "config.h"
+
+#include "TensorOneD.hpp"
+
+#include <math.h>
+#include <algorithm>
+#include <functional>
+#include <numeric>
+#include <utility>
+
+#include "geopm/Exception.hpp"
+
+namespace geopm
+{
+    TensorOneD::TensorOneD()
+    {
+    }
+
+    TensorOneD::TensorOneD(std::size_t dim)
+    {
+        set_dim(dim);
+    }
+
+    TensorOneD::TensorOneD(const TensorOneD &other)
+        : m_vec(other.m_vec)
+    {
+    }
+
+    TensorOneD::TensorOneD(TensorOneD &&other)
+        : m_vec(std::move(other.m_vec))
+    {
+    }
+
+    TensorOneD::TensorOneD(json11::Json input)
+    {
+        if (!input.is_array()) {
+            throw geopm::Exception("Neural network weights is non-array-type.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        if (input.array_items().size() == 0) {
+            throw geopm::Exception("Empty array is invalid for neural network weights.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        set_dim(input.array_items().size());
+
+        for (std::size_t idx = 0; idx < m_vec.size(); idx++) {
+            if (!input[idx].is_number()) {
+                throw geopm::Exception("Non-numeric type found in neural network weights.\n",
+                                       GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            m_vec[idx] = input[idx].number_value();
+        }
+    }
+
+    std::size_t TensorOneD::get_dim() const
+    {
+        return m_vec.size();
```

Comment:
Both `set_dim()` and `get_dim()` are one liners. They can be converted into *inline* functions by specifying the body in the `*.h` file, instead of the `*.cpp` file. It will save you an extra function call stack frame.  :)